submissions/api.py

```diff
@@ -764,6 +770,14 @@ def set_score(submission_uuid, points_earned, points_possible):
     try:
         score_model = score.save()
         _log_score(score_model)
+        if annotation_creator is not None:
+            score_annotation = ScoreAnnotation(
+                score=score_model,
+                creator=annotation_creator,
+                annotation_type=annotation_type,
+                reason=annotation_reason
+            )
+            score_annotation.save()
```

Comment:
I think it would make sense to validate the score annotation before saving the score, so that if there is an issue (say one of the field values being too long) it is caught first.